addons/middle-click-popup/addon.json

```
@@ -1,6 +1,6 @@
 {
   "name": "Insert blocks by name",
-  "description": "Middle click on the code area, use Ctrl+Space or Shift+Click to bring up a floating input box where you can type the name of a block (or parts of it) and drag the block into the code area. Hold Shift while dragging to avoid closing the box when adding multiple blocks at once.",
+  "description": "Middle click or Shift+Click on the code area, or press Ctrl+Space to bring up a block-search window, then type the name of a block (or parts of it) and drag it out of the window to insert it without finding it in the block palette. To add multiple blocks at once, hold Shift while dragging the blocks out to avoid closing the window.",
```

```suggestion
  "description": "Middle click, shift click, or press Ctrl+Space on the code area bring up a block search window, then type the name of a block (or parts of it) and drag it out of the window to insert it without finding it in the block palette. To add multiple blocks at once, hold Shift while dragging the blocks out to avoid closing the window.",
```

---

How can you activate a keyboard shortcut on something on the screen?

---

It was obvious to me, but it isn't to you it probably shouldn't be changed.

---

Should this mention something about the arrow keys and enter being able to be used in addition to dragging?

---

Another way we could achieve that is by showing icons of keys next to text indicating what different keys do on the block-search popup. Something like:
```
[Tab] Autocomplete  [Enter] Summon block
```

---

```suggestion
  "description": "Middle or shift click the code area or press Ctrl+Space to bring up a block search window. Type block names (or parts of them) and drag them onto the code area to add them to your project. The menu can be navigated with the arrow keys and enter and supports autocompleting with tab. To prevent the menu from closing, hold shift while dragging the blocks out of it",
```

---

Let's implement Samq's suggestion

---

But add a dot to the end of the last sentence

---

> But add a dot to the end of the last sentence

Replace "onto" with "in to"

---

```suggestion
  "description": "Middle or shift click the code area or press Ctrl+Space to bring up a block search window. Type block names (or parts of them) and drag them in to the code area to add them to your project. The menu can be navigated with the arrow keys and enter and supports autocompleting with tab. To prevent the menu from closing, hold shift while dragging blocks out of it.",
```

---

I think "in to" is grammatically incorrect - "into", all one word, seems to fit there better.

---

Also, keys should be capitalized (also with DNin01's suggestion):
```suggestion
  "description": "Middle or shift click the code area, or press Ctrl+Space to bring up a block search window. Type block names (or parts of them) and drag them into the code area to add them to your project. The menu can be navigated with the arrow keys and Enter and supports autocompleting with Tab. To prevent the menu from closing, hold Shift while dragging blocks out of it.",
```

---

Should "shift" in "shift click" be capitalized too?

---

> Should "shift" in "shift click" be capitalized too?

I don't know (though it probably should have a - or a + between shift and click).

---

> Should "shift" in "shift click" be capitalized too?

I think it's fine as-is

---

> I think "in to" is grammatically incorrect - "into", all one word, seems to fit there better.

Yeah, I knew that. I'm not sure whey I wrote it as two words there.